overlord/devicestate/devicestate.go

```diff
@@ -1574,6 +1587,48 @@ type CreateRecoverySystemOptions struct {
 
 var ErrNoRecoverySystem = errors.New("recovery system does not exist")
 
+func conflictingRecoverySystemChange(st *state.State) *state.Change {
+	for _, chg := range st.Changes() {
+		if chg.IsReady() {
+			continue
+		}
+		switch chg.Kind() {
+		case "create-recovery-system", "remove-recovery-system", "remodel":
+			return chg
+		}
+	}
+	return nil
+}
+
+// RemoveRecoverySystem removes the recovery system with the given label. The
+// current recovery system cannot be removed.
+func RemoveRecoverySystem(st *state.State, label string) (*state.Change, error) {
+	if chg := conflictingRecoverySystemChange(st); chg != nil {
+		return nil, fmt.Errorf("cannot remove recovery system while %q change with change id %q is in progress", chg.Kind(), chg.ID())
+	}
```

Comment:
we have dedicated errors and logic for this kind of thing in  snapstate/conflict.go

Reply:
we need the required conflicts also in the other direction, no remodel if there's an ongoing remove-recovery-system

Reply:
I converted things to use the errors from that package. Let me know if I should be moving some of the logic to `checkChangeConflictExclusiveKinds`. I didn't since these changes doesn't need to be fully exclusive, like the name suggests.

Reply:
`create-recovery-system` is checked for in `checkChangeConflictExclusiveKinds` already, so I'd update the code there and use it to keep things simple and consistent. If anyone complains we can revisit.